Sources/RemoteMessaging/RemoteMessagingStore.swift

```diff
@@ -25,11 +25,8 @@ import Persistence
 
 public enum RemoteMessagingStoreError: Error {
     case saveConfigFailed
-    case invalidateConfigFailed
     case updateMessageShownFailed
-    case saveMessageFailed
     case updateMessageStatusFailed
-    case deleteScheduledMessageFailed
```

Comment:
@ayoy I'm fine to simplify overall, though one risk is that now we have multiple different functions that can emit the same error with no way to disambiguate between them, which could hurt us when we get an anomaly to investigate. Unless the error parameter is enough to go on?

Reply:
I'm not simplifying this because I want to, but because I'm forced to do it because there will be less context saves. Previously we've had separate save for storing new config and for storing a message. This breaks transactionality, putting us at risk of having a new config but an old message. My change removes extra saves so that each action performed on the store from the outside is a database transaction with a single context save. We'll still be getting a Core Data context save error and we'll be able to debug it further. My point is that the previous implementation was kind of broken in that regard. Wdyt?

Reply:
I see, I think it makes sense in that case, let's go with it. Thanks!